omniscidb/QueryEngine/Compiler/genx.ll

```diff
@@ -44,41 +60,52 @@ define i8 @thread_warp_idx(i8 noundef %warp_sz) {
     ret i8 0
 }
 
-define i64 @agg_count(i64* %agg, i64 %val) {
-    %ld = load i64, i64* %agg
-    %old = atomicrmw add i64* %agg, i64 1 monotonic
+define i64 @agg_count_shared(i64 addrspace(4)* %agg, i64 noundef %val) {
+    %ld = load i64, i64 addrspace(4)* %agg
+    %old = atomicrmw add i64 addrspace(4)* %agg, i64 1 monotonic
     ret i64 %old
 }
 
-define i64 @agg_count_skip_val(i64* %agg, i64 noundef %val, i64 noundef %skip_val) {
+define i32 @agg_count_int32_shared(i32 addrspace(4)* %agg, i32 noundef %val) {
+    %ld = load i32, i32 addrspace(4)* %agg
+    %old = atomicrmw add i32 addrspace(4)* %agg, i32 1 monotonic
+    ret i32 %old
+}
+
+define i64 @agg_count_skip_val_shared(i64 addrspace(4)* %agg, i64 noundef %val, i64 noundef %skip_val) {
     %no_skip = icmp ne i64 %val, %skip_val
     br i1 %no_skip, label %.noskip, label %.skip
 .noskip:
-    %old = call i64 @agg_count(i64* %agg, i64 %val)
```

Comment:
What happens if `shared_mem` is turned off? What function implementation do we fall back to?

Reply:
Just noticed the comment. I haven't tested it, but what should happen is it should emit the `agg_count` from the "cpu" runtime module instead. BTW, I'm not sure if we need the option to disable shared memory at all :)